src/constraints/constraint_storage_lifetime.jl

```diff
@@ -33,10 +33,22 @@ function add_constraint_storage_lifetime!(m::Model)
                 for (n, s, t) in storages_invested_available_indices(m; node=n, stochastic_scenario=s, t=t);
                 init=0,
             )
-            >=
-            sum(
-                storages_invested[n, s_past, t_past]
-                for (n, s_past, t_past) in _past_storages_invested_available_indices(m, n, s, t)
+            >= #FIXME: >= for econ lifetime, <= for tech lifetime
+            + sum(
+                + storages_invested[n, s_past, t_past]
+                * storage_capacity_transfer_factor[(node=n, stochastic_structure__stochastic_scenario=s_past,vintage_t=first(t_past.start),t=t)]
+                for (n, s_past, t_past) in storages_invested_available_indices(
```

Comment:
How come stochastic_structure__stochastic_scenario is one of the dimensions? That doesn't sound right. Also, as said above, accessing a field (t_past.start) is asking for trouble, better use start(t_past)